src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCoalescingAssignmentTests.cs

```diff
@@ -733,14 +800,181 @@ .maxstack  3
+                //         o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_ByReferenceVariableMustBeInitialized, "").WithLocation(9, 22),
+                // (10,13): error CS0118: 'o1' is a variable but is used like a type
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_BadSKknown, "o1").WithArguments("o1", "variable", "type").WithLocation(10, 13),
+                // (10,16): error CS1001: Identifier expected
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "??=").WithLocation(10, 16),
+                // (10,16): error CS1002: ; expected
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "??=").WithLocation(10, 16),
+                // (10,16): error CS1525: Invalid expression term '??='
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "??=").WithArguments("??=").WithLocation(10, 16),
+                // (10,16): error CS8174: A declaration of a by-reference variable must have an initializer
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_ByReferenceVariableMustBeInitialized, "").WithLocation(10, 16),
+                // (10,20): error CS1525: Invalid expression term 'ref'
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "ref").WithArguments("ref").WithLocation(10, 20),
+                // (10,20): error CS1002: ; expected
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "ref").WithLocation(10, 20),
+                // (10,24): error CS0118: 'o2' is a variable but is used like a type
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_BadSKknown, "o2").WithArguments("o2", "variable", "type").WithLocation(10, 24),
+                // (10,26): error CS1001: Identifier expected
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, ";").WithLocation(10, 26),
+                // (10,26): error CS8174: A declaration of a by-reference variable must have an initializer
+                //         ref o1 ??= ref o2;
+                Diagnostic(ErrorCode.ERR_ByReferenceVariableMustBeInitialized, "").WithLocation(10, 26)
+            });
+        }
+
+        [Fact]
+        public void InParameter()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        object o = null;
+        M(o ??= ""Test String"");
+    }
+
+    static void M(in object o) => Console.WriteLine(o);
+}";
+
+            var verifier = CompileAndVerify(source, expectedOutput: "Test String");
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       19 (0x13)
+  .maxstack  2
+  .locals init (object V_0)
+  IL_0000:  ldnull
+  IL_0001:  dup
```

Comment:
May want to consider having `o` be a parameter vs. a local. The way the code is written it loosk like we're essentially optimizing away `o` in the test which doesn't seem to line up with the intent.